app/controllers/application_controller.rb

```diff
@@ -2,12 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
-  before_filter :configure_permitted_parameters, if: :devise_controller?
+  before_filter :configure_devise_permitted_parameters, if: :devise_controller?
 
   protected
 
-  def configure_permitted_parameters
-    devise_parameter_sanitizer.for(:sign_up) << [:first_name, :last_name]
-    devise_parameter_sanitizer.for(:sign_up) << :profile_photo
+  def configure_devise_permitted_parameters
+    registration_params = [
+      :first_name, :last_name, :email, :password,
+      :password_confirmation, :profile_photo
+    ]
+
+    if params[:action] == 'update'
+      devise_parameter_sanitizer.for(:account_update) do |u|
+        u.permit(registration_params << :current_password)
+      end
+    elsif params[:action] == 'create'
+      devise_parameter_sanitizer.for(:sign_up) do |u|
+        u.permit(registration_params)
+      end
```

Comment:
This method is pretty long. Why not divide it into 3: a method that sets registration_params, and then two that handle whatever is needed in either state?